### DISCUSSION TITLE
Multiple input types with LLM, which encoders?

### DISCUSSION THREAD

**Discussion by adivoj** [ludwig-ai/ludwig#3620] [General]
Hi and thanks for Ludwig!

I have text input which I want llama2 (auto_transformer) to handle and I have 5 signed number arrays. My output is a number from 1 to 10.

Should the number arrays be of type number or timeseries? They are kind of timeseries but normalized. What should the encoder be there?

What should the output type be, category or regression?

Thanks!

**Comment by tgaddair** [comment]
Hey @adivoj,

For the input features, since it sounds like they're lists of numbers, I would choose the [Timeseries](https://ludwig.ai/latest/configuration/features/time_series_features/) data type. Using the default (parallel CNN) encoder is probably a reasonable place to start.

For the output feature, I would usually say that if you are okay with the output value being continuous / a real number, then you can make the data type Number (regression), but if you want it to be discrete / an integer, then you can choose Category. If you choose Category as the output type and the ordering matters, I would also suggest using the CORN loss (apologies it's not documented for some reason, but the config can be found [here](https://github.com/ludwig-ai/ludwig/blob/master/ludwig/schema/features/loss/loss.py#L447)).
